apps/studio/src/schemas/resource.ts

```diff
@@ -70,3 +71,18 @@ export const getAncestrySchema = z.object({
   siteId: z.string(),
   resourceId: z.string().optional(),
 })
+
+export const searchSchema = z
+  .object({
+    siteId: z.string(),
+    query: z.string().optional(),
+  })
+  .merge(infiniteOffsetPaginationSchema)
+
+export const searchOutputSchema = z.object({
+  totalCount: z.number().nullable(),
+  resources: z.array(z.custom<SearchResultResource>()),
+  suggestions: z.object({
+    recentlyEdited: z.array(z.custom<SearchResultResource>()),
+  }),
```

Comment:
question: will there be other kinds of suggestions in the future?

Reply:
no such explicit product or design requirement

but i remember during designcrit there seemed to be a common theme and questions around supporting other form of suggestions e.g. non-resource pages like settings, or isomer guide

was also taking inspiration from other search systems e.g. Notion, Slack, and they seem to support other form of suggestions. Or even filters

thus decided to make the call to pre-empt that and structure it this way

Reply:
Hmm I feel that we shouldn't be doing this pre-emptively especially when this is more of an internal API that is relatively easy to change. Having this means that we will definitely support providing other kinds of suggestions (which may or may not require a backend call to retrieve), but we don't have a foreseeable timeline yet regarding whether we will actually add such a support.

Reply:
okay makes, sense, shall do!